src/Coordination/KeeperStateMachine.cpp

```diff
@@ -203,12 +204,15 @@ void KeeperStateMachine::create_snapshot(
         {
             {   /// Read storage data without locks and create snapshot
                 std::lock_guard lock(snapshots_lock);
-                auto snapshot_buf = snapshot_manager.serializeSnapshotToBuffer(*snapshot);
-                auto result_path = snapshot_manager.serializeSnapshotBufferToDisk(*snapshot_buf, snapshot->snapshot_meta->get_last_log_idx());
-                latest_snapshot_buf = snapshot_buf;
+                auto [path, error_code]= snapshot_manager.serializeSnapshotToDisk(*snapshot);
+                if (error_code)
+                {
+                    throw Exception(ErrorCodes::LOGICAL_ERROR, "Snapshot {} was created failed, error: {}",
```

Comment:
```suggestion
                    throw Exception(ErrorCodes::LOGICAL_ERROR, "Failed to create snapshot {} , error code: {}",
```
BTW, why is it `LOGICAL_ERROR`?

Reply:
The directory should be writeable or enough space, but error happens, so I thought it was a LOGIC_ERROR.

Reply:
What's your advice?

Reply:
Should be IO error, I will change.